src/app/profile/profile.component.ts

```diff
@@ -0,0 +1,108 @@
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormControl, FormArray, Validators } from '@angular/forms';
+import { customEmailValidator } from './validators/email.validator';
+
+import { Profile } from './profile.model';
+import { ProfileService } from '../profile.service';
+
+@Component({
+  selector: 'app-profile',
+  templateUrl: './profile.component.html',
+  styleUrls: ['./profile.component.css']
+})
+
+export class ProfileComponent implements OnInit {
+
+  profiles: any;
+
+  profileForm!: FormGroup;
+
+  firstName = new FormControl('', Validators.required);
+  lastName = new FormControl('', Validators.required);
+  email = new FormControl('', [ Validators.required, customEmailValidator() ]);
+
+  constructor(private profileService: ProfileService) { }
+
+  getProfileData(): void {
+    this.profileService.getProfile().subscribe(profiles => this.profiles = profiles);
+  }
+
+  ngOnInit(): void {
+    this.getProfileData();
+
+    this.profileForm = new FormGroup({
+      firstName: this.firstName,
+      lastName: this.lastName,
+      email: this.email,
+      phoneNumbers: new FormArray([])
+    });
+
+    this.populatePhoneFieldsBasedOnData();
```

Comment:
you have a race condition here.  this.getProfileData() is subscribing to an observable, and observables are not guaranteed to emit a value right away.  For example, if you're fetching data over the network to an API, its going to take time and your subscribe call back will fire later.  So while you're data is in transit, you're already starting to build your form and you're going to get nailed with an undefined error.

My suggestion is to toss your form building code into it's own function called "buildForm(profile: Profile)".  Then in your subscribe instead of:

`this.profileService.getProfile().subscribe(profile => this.profile = profile);`

you do 

`this.profileService.getProfile().subscribe(profile => buildForm(profile));`

you're going to need an *ngIf in the profile to hide the contents until the form as been built, or you might get in trouble there too since the template will probably render before the data arrives.

This will also reduce the amount of state needed in your component.

lastly, when you subscribe to observables, you need to make sure you have an unsubscribe mechanism if its a hot observable.  Otherwise you risk a memory leak.  Ask me about this later.